txpipe/data_types/base.py

```diff
@@ -444,4 +445,28 @@ def write_provenance(self):
         pass
 
     def read_provenance(self):
-        raise ValueError("Reading existing PNG files is not supported")
\ No newline at end of file
+        raise ValueError("Reading existing PNG files is not supported")
+
+class PickleFile(DataFile):
+    suffix = "pkl"
+
+    @classmethod
+    def open(self, path, mode, **kwargs):
+        return open(path, mode + "b")
```

Comment:
should self.mode be updated?

Reply:
I'd like to keep this as "r" or "w" throughout, as below.  This is consistent with what we do with FITS files.